pkg/ccl/backupccl/backup_test.go

```diff
@@ -1411,177 +1411,85 @@ into_db='restoredb', %s)`, encryptionOption), backupLoc1)
 	}
 }
 
-type inProgressChecker func(context context.Context, ip inProgressState) error
-
-// inProgressState holds state about an in-progress backup or restore
-// for use in inProgressCheckers.
-type inProgressState struct {
-	*gosql.DB
-	backupTableID uint32
-	dir, name     string
-}
-
-func (ip inProgressState) latestJobID() (jobspb.JobID, error) {
-	var id jobspb.JobID
-	if err := ip.QueryRow(
-		`SELECT job_id FROM crdb_internal.jobs ORDER BY created DESC LIMIT 1`,
-	).Scan(&id); err != nil {
-		return 0, err
-	}
-	return id, nil
-}
+func TestBackupJobUpdatesProgress(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	defer jobs.TestingSetProgressThresholds()()
 
-// checkInProgressBackupRestore will run a backup and restore, pausing each
-// approximately halfway through to run either `checkBackup` or `checkRestore`.
-func checkInProgressBackupRestore(
-	t testing.TB, checkBackup inProgressChecker, checkRestore inProgressChecker,
-) {
-	var allowResponse chan struct{}
-	var exportSpanCompleteCh chan struct{}
+	blockCh := make(chan struct{})
 	params := base.TestClusterArgs{}
 	knobs := base.TestingKnobs{
 		DistSQL: &execinfra.TestingKnobs{
 			BackupRestoreTestingKnobs: &sql.BackupRestoreTestingKnobs{
-				RunAfterExportingSpanEntry: func(_ context.Context, res *kvpb.ExportResponse) {
-					<-allowResponse
-					// If ResumeSpan is set to nil, it means that we have completed
-					// exporting a span and the job will update its fraction progressed.
-					if res.ResumeSpan == nil {
-						<-exportSpanCompleteCh
-					}
-				},
-				RunAfterProcessingRestoreSpanEntry: func(_ context.Context, _ *execinfrapb.RestoreSpanEntry) {
-					<-allowResponse
+				RunAfterExportingSpanEntry: func(_ context.Context, _ *kvpb.ExportResponse) {
+					<-blockCh
 				},
 			},
 		},
 		JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
 	}
 	params.ServerArgs.Knobs = knobs
 
-	const numAccounts = 100
-
-	ctx := context.Background()
-	_, sqlDB, dir, cleanup := backupRestoreTestSetupWithParams(t, multiNode, numAccounts,
+	_, sqlDB, _, cleanupFn := backupRestoreTestSetupWithParams(t, singleNode, 1,
 		InitManualReplication, params)
-	conn := sqlDB.DB.(*gosql.DB)
-	defer cleanup()
-
-	sqlDB.Exec(t, `CREATE DATABASE restoredb`)
 	// the small test-case will get entirely buffered/merged by small-file merging
 	// and not report any progress in the meantime unless it is disabled.
 	sqlDB.Exec(t, `SET CLUSTER SETTING bulkio.backup.file_size = '1'`)
 	sqlDB.Exec(t, `SET CLUSTER SETTING bulkio.backup.merge_file_buffer_size = '1'`)
```

Comment:
[q] Does this setting actually do anything anymore? No need to fix in this PR, but perhaps we should follow up with something that rips this out of our tests if it doesn't do anything.

Reply:
good point, will rip this out